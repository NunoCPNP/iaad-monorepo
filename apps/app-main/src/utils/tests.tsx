import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@emotion/react'

import { dark } from '../styles/themes/dark'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={dark}>{children}</ThemeProvider>
}

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
