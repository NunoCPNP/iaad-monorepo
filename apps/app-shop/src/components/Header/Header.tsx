import { Container } from './Header.styles'

import NavBar from '../NavBar/NavBar'

type Props = {
  navbar: any
}

const Header: React.FC<Props> = ({ navbar }) => (
  <Container data-testid="header">
    <NavBar navbar={navbar} />
  </Container>
)

export default Header
