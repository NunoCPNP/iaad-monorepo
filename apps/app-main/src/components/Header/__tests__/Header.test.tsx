import { render, cleanup } from '../../../utils/tests'

import Header from '../Header'

afterEach(cleanup)

describe('<Header /> spec', () => {
  it('Should run without errors', () => {
    const { queryAllByTestId } = render(<Header />)

    const header = queryAllByTestId(/header/i)
    expect(header).toHaveLength(1)
  })

  it('Should match snapshot', () => {
    const { asFragment } = render(<Header />)

    expect(asFragment(<Header />)).toMatchSnapshot()
  })
})
