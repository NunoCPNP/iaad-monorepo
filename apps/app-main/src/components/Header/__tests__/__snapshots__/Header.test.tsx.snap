// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> spec Should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-position: sticky;
  position: sticky;
  top: 0;
  z-index: 100;
}

.emotion-1 {
  width: 100%;
  background-color: #C1292E;
  box-shadow: 0px 1px 8px 0px rgba(0, 0, 0, 0.3);
  -webkit-transition: 0.3s ease background-color;
  transition: 0.3s ease background-color;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  max-width: 140rem;
  height: 6rem;
  margin: 0 auto;
}

@media only screen and (max-width: 1400px) {
  .emotion-2 {
    padding: 0 1rem;
  }
}

.emotion-3 {
  width: 4rem;
}

.emotion-4 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 li a {
  font-family: 'Roboto',sans-serif;
  text-transform: uppercase;
  margin-left: 1.5rem;
  font-size: 1.3rem;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FCFAFA;
  -webkit-transition: 0.2s ease-out color;
  transition: 0.2s ease-out color;
}

.emotion-4 li a:hover {
  color: #272727;
}

@media only screen and (max-width: 768px) {
  .emotion-4 {
    display: none;
  }
}

.emotion-5 {
  font-size: 50%;
  width: 6em;
  height: 4.5em;
  position: relative;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.5s ease-in-out;
  transition: 0.5s ease-in-out;
  cursor: pointer;
}

.emotion-5 span {
  display: block;
  position: absolute;
  height: 0.6em;
  width: 50%;
  background: #FCFAFA;
  opacity: 1;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.25s ease-in-out;
  transition: 0.25s ease-in-out;
}

.emotion-5 span:nth-of-type(even) {
  left: 50%;
  border-radius: 0 0.9em 0.9em 0;
}

.emotion-5 span:nth-of-type(odd) {
  left: 0em;
  border-radius: 0.9em 0 0 0.9em;
}

.emotion-5 span:nth-of-type(1),
.emotion-5 span:nth-of-type(2) {
  top: 0em;
}

.emotion-5 span:nth-of-type(3),
.emotion-5 span:nth-of-type(4) {
  top: 1.8em;
}

.emotion-5 span:nth-of-type(5),
.emotion-5 span:nth-of-type(6) {
  top: 3.6em;
}

.emotion-5.open span:nth-of-type(1),
.emotion-5.open span:nth-of-type(6) {
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.emotion-5.open span:nth-of-type(2),
.emotion-5.open span:nth-of-type(5) {
  -webkit-transform: rotate(-45deg);
  -moz-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.emotion-5.open span:nth-of-type(1) {
  left: 0.5em;
  top: 0.7em;
}

.emotion-5.open span:nth-of-type(2) {
  left: calc(50% - 0.5em);
  top: 0.7em;
}

.emotion-5.open span:nth-of-type(3) {
  left: -50%;
  opacity: 0;
}

.emotion-5.open span:nth-of-type(4) {
  left: 100%;
  opacity: 0;
}

.emotion-5.open span:nth-of-type(5) {
  left: 0.5em;
  top: 2.9em;
}

.emotion-5.open span:nth-of-type(6) {
  left: calc(50% - 0.5em);
  top: 2.9em;
}

@media only screen and (min-width: 769px) {
  .emotion-5 {
    display: none;
  }
}

<header
    class="emotion-0"
    data-testid="header"
  >
    <nav
      class="emotion-1"
      data-testid="navbar"
    >
      <div
        class="emotion-2"
      >
        <div>
          <div
            class="emotion-3"
          >
            <div
              style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 200px; height: 50px;"
            >
              <img
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
        </div>
        <div
          class="emotion-4"
        >
          <ul
            style="opacity: 1; transform: scale(0) translateZ(0);"
          >
            <li
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <a
                href="/"
              >
                common:navbar.home
              </a>
            </li>
            <li
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <a
                href="/#who"
              >
                common:navbar.who
              </a>
            </li>
            <li
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <a
                href="/#services"
              >
                common:navbar.services
              </a>
            </li>
            <li
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <a
                href="/#contacts"
              >
                common:navbar.contacts
              </a>
            </li>
            <li
              style="opacity: 0; transform: translateY(20px) translateZ(0);"
            >
              <a
                href="/#ctc"
              >
                common:navbar.ctc
              </a>
            </li>
          </ul>
        </div>
        <div
          class=" emotion-5"
          data-testid="burgericon"
        >
          <span />
          <span />
          <span />
          <span />
          <span />
          <span />
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
